bokeh/state.py

```diff
@@ -0,0 +1,516 @@
+                One of: 'inline', 'cdn', 'relative(-dev)' or 'absolute(-dev)'.
+                See :class:`bokeh.resources.Resources` for more details.
+
+            root_dir (str, optional) :
+
+        .. note::
+            Generally, this should be called at the beginning of an interactive
+            session or the top of a script.
+
+        .. warning::
+            This output file will be overwritten on every save, e.g., each time
+            show() or save() is invoked, or any time a Bokeh plotting API
+            causes a save, if ``autosave`` is True.
+
+        """
+        self._file = {
+            'filename'  : filename,
+            'resources' : Resources(mode=mode, root_dir=root_dir),
+            'autosave'  : autosave,
+            'title'     : title,
+        }
+
+        if os.path.isfile(filename):
+            logger.info("Session output file '%s' already exists, will be overwritten." % filename)
+
+    def output_notebook(self, url=None, docname=None, session=None, name=None):
+        self._notebook = True
+
+        if session or url or name:
+            if docname is None:
+                docname = "IPython Session at %s" % time.ctime()
+            self.output_server(docname, url=url, session=session, name=name)
+
+    def output_server(self, docname, session=None, url="default", name=None, clear=True):
+        """ Store Bokeh plots and objects on a Bokeh server.
+
+        Args:
+            docname (str) : Name of document to push on Bokeh server
+                Any existing documents with the same name will be overwritten.
+
+            session (Session, optional) : An explicit session to use (default: None)
+                If None, a new default session is created.
+
+            url (str, optional) : URL of the Bokeh server (default: "default")
+                If "default", then ``session.DEFAULT_SERVER_URL`` is used.
+
+            name (str, optional) : A name for the session
+                If None, the server URL is used as the name
+
+            clear (bool, optional) : Whether to clear the document (default: True)
+                If True, an existing server document will be cleared of any
+                existing objects.
+
+        Returns:
+            None
+
+        .. note::
+            Generally, this should be called at the beginning of an interactive
+            session or the top of a script.
+
+        .. warning::
+            Calling this function will replace any existing default session.
+
+        """
+        if url == "default":
+            url = DEFAULT_SERVER_URL
+
+        if name is None:
+            name = url
+
+        if not session:
+            self._session = Session(name=name, root_url=url)
+        else:
+            self._session = session
+
+        self._session.use_doc(docname)
+        self._session.load_document(self._document)
+
+        if clear:
+            self._document.clear()
+
```

Comment:
OK, you have the new method: `output_file`, `output_notebook` and `output_server`... do you want to document it in the same way you do it outside the class? Maybe a simple docstring is enough and you get the full documentation in the functions below... or even better... maybe you can, in some way, documents these methods and make the functions replicate the documentation from the methods... if not there is a duplication of information an both docs can be unsynced in the future...